spanner/google/cloud/spanner/database.py

```diff
@@ -79,6 +79,7 @@ def __init__(self, database_id, instance, ddl_statements=(), pool=None):
         self.database_id = database_id
         self._instance = instance
         self._ddl_statements = _check_ddl_statements(ddl_statements)
+        self._transaction_running = False
```

Comment:
@jonparrott just gave me the exact same idea. Note that it will not work if the user uses some other concurrency strategy though.